src/main/java/org/openrewrite/github/ReplaceRunners.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2023 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.github;
+
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import org.openrewrite.Option;
+import org.openrewrite.Recipe;
+import org.openrewrite.internal.lang.NonNull;
+import org.openrewrite.yaml.ChangeValue;
+
+import java.time.Duration;
+import java.util.Arrays;
+import java.util.List;
+
+@EqualsAndHashCode(callSuper = false)
+@Getter
+public class ReplaceRunners extends Recipe {
+    @Option(displayName = "Job Name",
+            description = "The name of the job to update",
+            example = "build")
+    private final String jobName;
+
+    @Option(displayName = "Runners",
+            description = "The new list of runners to set",
+            example = "runner1,runner2"
+    )
+    private final List<String> runners;
+
+    public ReplaceRunners(@NonNull String jobName, @NonNull List<String> runners) {
+        this.jobName = jobName;
+        this.runners = runners;
+        doNext(new ChangeValue(
+                String.format("$.jobs.%s.runs-on",jobName),
+                Arrays.toString(runners.toArray()),
+                ".github/workflows/*.yml")
+        );
+    }
+
```

Comment:
For future references, is a `List` OK in terms of what rewrite allows for input? Or should it be a String instead? 

Also, what's the recommended way to do validation? For example, `jobName` should not be null, and `runners` should not be null or empty

Reply:
To answer your validation question I quickly looked at how other recipes use implicit and explicit validation.
[`AddCronTrigger.cron`](https://github.com/openrewrite/rewrite-github-actions/blob/main/src/main/java/org/openrewrite/github/AddCronTrigger.java#L32) for instance is `final`, which ensures it's at least defined, and typically not empty.
While [`AutoCancelInProgressWorkflow.accessToken`](https://github.com/openrewrite/rewrite-github-actions/blob/main/src/main/java/org/openrewrite/github/AutoCancelInProgressWorkflow.java#L41) has an explicit `org.openrewrite.internal.lang.Nullable` annotation to mark it optional, as well as being non-final.

As for the `List` question; I found an Option using a List in [`DeclarationSiteTypeVariance.variantTypes`](https://github.com/openrewrite/rewrite/blob/v7.35.0/rewrite-java/src/main/java/org/openrewrite/java/cleanup/DeclarationSiteTypeVariance.java#L39), [configured through yaml](https://github.com/openrewrite/rewrite/blob/v7.35.0/rewrite-java/src/main/resources/META-INF/rewrite/cleanup.yml#L58).

Does that answer your questions?

Reply:
Perfect: I've moved it to Ready for Review; Thanks again!